pages/sign-in.js

```diff
@@ -1,85 +1,130 @@
-import React from 'react';
+import React, { useState } from 'react';
 import Image from 'next/image';
 import Link from 'next/link';
+import { useForm } from 'react-hook-form';
+import { yupResolver } from '@hookform/resolvers/yup';
+import * as yup from 'yup';
 import { Button, Input, MnavBar } from '@/components/index';
 
 import { LogoBlack, SignInImage } from '../assets';
 import { PasswordCheckIcon } from '../assets/svgs/svgs';
 
 const SignIn = () => {
+  const schema = yup
+    .object({
+      email: yup.string().email('Kindly provide a valid email address').required('Email is required'),
+      password: yup.string().required('password is required').min(6, 'Password must be atleast 6 character long')
+    })
+    .required();
+
+  const {
+    register,
+    handleSubmit,
+    formState: { errors }
+  } = useForm({
+    resolver: yupResolver(schema)
+  });
+
+  const onSubmit = (data) => {
+    alert(JSON.stringify(data));
+  };
+  const [showPassword, setShowPassword] = useState(false);
```

Comment:
This logic should be inside the `Input` component. This will ensure that the `password` field is consistent across the application.